client/azure_client_set.go

```diff
@@ -11,12 +11,194 @@ import (
+	)
+
+	storageAccountsClientMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_storageaccounts",
+			Help:       "HTTP metrics for Azure SDK Storage Accounts client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+
+	usageClientMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_usage",
+			Help:       "HTTP metrics for Azure SDK Usage client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+
+	virtualNetworkMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_virtualnetwork",
+			Help:       "HTTP metrics for Azure SDK Virtual Network client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+
+	virtualNetworkGatewaysMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_virtualnetworkgateways",
+			Help:       "HTTP metrics for Azure SDK Virtual Network Gateways client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+
+	virtualNetworkGatewayConnectionsMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_virtualnetworkgatewayconnections",
+			Help:       "HTTP metrics for Azure SDK Virtual Network Gateway Connections client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+
+	virtualMachineScaleSetsMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_virtualmachinescalesets",
+			Help:       "HTTP metrics for Azure SDK Virtual Machine ScaleSets client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+
+	virtualMachineScaleSetVMsMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_virtualmachinescalesetvms",
+			Help:       "HTTP metrics for Azure SDK Virtual Machine ScaleSet VMs client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+
+	vnetPeeringMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_vnetpeering",
+			Help:       "HTTP metrics for Azure SDK VNET Peering client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"status_code"},
+	)
+)
+
+func init() {
+	// Register prometheus metrics.
+	err := prometheus.Register(deploymentsClientMetricsDesc)
+	if err != nil && !collector.IsAlreadyRegisteredError(err) {
+		panic(err)
+	}
+	err = prometheus.Register(groupsClientMetricsDesc)
+	if err != nil && !collector.IsAlreadyRegisteredError(err) {
+		panic(err)
```

Comment:
Let’s add microerror.JSON/microerror.Stack here.

Reply:
What do you mean with `microerror.Stack`? That does not exist (anymore).

Reply:
I think this codebase still uses v0.1.0